packages/jest-haste-map/src/index.js

```diff
@@ -0,0 +1,241 @@
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+const crypto = require('crypto');
+const denodeify = require('denodeify');
+const docblock = require('./lib/docblock');
+const execSync = require('child_process').execSync;
+const fs = require('graceful-fs');
+const getPlatformExtension = require('./lib/getPlatformExtension');
+const nodeCrawl = require('./crawlers/node');
+const os = require('os');
+const path = require('./fastpath');
+const watchmanCrawl = require('./crawlers/watchman');
+
+const GENERIC_PLATFORM = 'generic';
+const NODE_MODULES = path.sep + 'node_modules' + path.sep;
+const PACKAGE_JSON = path.sep + 'package.json';
+const VERSION = require('../package.json').version;
+
+const readFile = denodeify(fs.readFile);
+const writeFile = denodeify(fs.writeFile);
+
+const canUseWatchman = (() => {
+  try {
+    execSync('watchman version', {stdio: 'ignore'});
+    return true;
+  } catch (e) {}
+  return false;
+})();
+
+class HasteMap {
+
+  constructor(options) {
+    this._options = {
+      cacheDirectory: options.cacheDirectory || os.tmpDir(),
+      extensions: options.extensions,
+      ignorePattern: options.ignorePattern,
+      platforms: options.platforms,
+      resetCache: options.resetCache,
+      roots: options.roots,
+      useWatchman:
+        options.useWatchman === undefined ? true : options.useWatchman,
+    };
+
+    const list = options.providesModuleNodeModules;
+    this._whitelist = (list && list.length)
+      ? new RegExp('(.+?)' + NODE_MODULES + '(' + list.join('|') + ')(?!' + NODE_MODULES + ')')
+      : null;
+
+    this._buildPromise = null;
+    this._cachePath = HasteMap.getCacheFilePath(
+      this._options.cacheDirectory,
+      VERSION,
+      this._options.roots.join(':'),
+      this._options.extensions.join(':'),
+      this._options.platforms.join(':')
+    );
+  }
+
+  static getCacheFilePath(tmpdir) {
+    const hash = crypto.createHash('md5');
+    Array.from(arguments).slice(1).forEach(arg => hash.update(arg));
+    return path.join(tmpdir, hash.digest('hex'));
+  }
+
+  build() {
+    if (!this._buildPromise) {
+      this._buildPromise = this._buildFileMap()
+        .then(data => this._buildHasteMap(data));
+    }
+    return this._buildPromise;
+  }
+
+  persist(data) {
+    return writeFile(this._cachePath, JSON.stringify(data)).then(() => data);
+  }
+
+  read() {
+    return this._parse(fs.readFileSync(this._cachePath, 'utf-8'));
+  }
+
+  matchFiles(pattern) {
+    if (!(pattern instanceof RegExp)) {
+      pattern = new RegExp(pattern);
+    }
+    return this.build().then(data => {
+      const files = [];
+      for (const file in data.files) {
+        if (pattern.test(file)) {
+          files.push(file);
+        }
+      }
+      return files;
+    });
+  }
+
+  _buildFileMap() {
+    const read = this._options.resetCache ? this._createEmptyMap : this.read;
+
+    return Promise.resolve()
+      .then(() => read.call(this))
```

Comment:
`Promise.resolveI(read.call(this))`

or

`Promise.resolve(this._options.resetCache ? this._createEmptyMap() : this.read())`